### PR TITLE
feat: add icon list into schema

### PR DESCRIPTION
**Pull request by supproduction**
splits data-clit into revealers
adds icon list into schema

closes [HRZ-2889](https://spryker.atlassian.net/browse/HRZ-2889)

[HRZ-2889]: https://spryker.atlassian.net/browse/HRZ-2889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/domain/content/link/src/link.schema.ts [hunk 1]
```diff
@@ -69,28 +70,11 @@ export const linkComponentSchema: ContentComponentSchema<ContentLinkComponent> =
...
-          {
-            value: 'mark',
-            text: 'mark',
-          },
-        ],
+        options:
+          iconInjectable
+            .get()
+            ?.getIcons()
+            .map((i) => ({ value: i, text: i })) ?? [],
```
Comment (tobi-or-not-tobi) [inline libs/domain/content/link/src/link.schema.ts:78]: what about sorting them?

File: libs/domain/search/src/services/revealers/products-experience-data.revealer.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  ExperienceDataRevealer,
+  MessageType,
+  postMessage,
+} from '@spryker-oryx/experience';
+import { Observable, of, switchMap, tap } from 'rxjs';
+import { Suggestion } from '../../models';
+
+import { SuggestionService } from '../suggestion';
+
+export class ProductsExperienceDataRevealer implements ExperienceDataRevealer {
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/src/services/revealers/products-experience-data.revealer.ts:13]: I've tried to use the product sku editor inside EB and it no longer works. not sure if it's related to this PR though.

File: libs/domain/search/src/services/search.providers.ts [hunk 3]
```diff
@@ -44,3 +46,10 @@ export const searchProviders: Provider[] = [
     useClass: SearchPageTitleMetaResolver,
   },
 ];
+
+export const searchPreviewProviders: Provider[] = [
+  {
+    provide: ExperienceDataRevealer,
```
Comment (tobi-or-not-tobi) [inline libs/domain/search/src/services/search.providers.ts:52]: I prefer to not hang the revealer up on Experience Builder, as other integrations can benefit from this as well.

Comment (tobi-or-not-tobi) [reply libs/domain/search/src/services/search.providers.ts:52]: I do like this revealer pattern though!

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: libs/domain/search/src/services/revealers/products-experience-data.revealer.ts (hunk 2 above)
Comment (supproduction) [reply libs/domain/search/src/services/revealers/products-experience-data.revealer.ts:13]: It's not releated to this PR. we have bug in EB

**Review by tobi-or-not-tobi — APPROVED** [review]

**Pull request merged**